### PR TITLE
Use k8s cload creeated by add-k8s --local for bootstrapping

### PR DESCRIPTION
**Pull request by ycliuhw**
## Description of change

Set the cloud-name in JujuData for bootstrapping to k8s cloud created by add-k8s --local 

## QA steps

To run a test locally with lxd and locally complied juju:

  * ```$ mkdir /tmp/test-run```
  * ```$ export JUJU_HOME=/tmp/test-run```
  * ```$ vim $JUJU_HOME/environments.yaml  # copy gke & microk8s environment from cloud-city```
    ```yaml
    environments:
        gke:
            type: kubernetes
            test-mode: true
            ...
        microk8s:
            type: kubernetes
            test-mode: true
            ...
       
    ```
  * ```export JUJU_REPOSITORY=$GOPATH/src/github.com/juju/juju/acceptancetests/repository```
  * ```mkdir /tmp/artifacts```
  * Now you can run the test with:
     * ```$ ./acceptancetests/assess_caas_deploy_charms.py gke $GOPATH/bin/juju /tmp/artifacts nw-caas-deploy-charms-gke --caas-image-repo=jujuqabot --caas-provider=GKE --k8s-controller --debug```
    * ```$ ./acceptancetests/assess_caas_deploy_charms.py microk8s $GOPATH/bin/juju /tmp/artifacts nw-caas-deploy-charms-microk8s --caas-image-repo=jujuqabot --caas-provider=MICROK8S --k8s-controller --debug```

## Documentation changes

None

## Bug reference

None

### CODE REVIEW

**Comment by ycliuhw** [comment]
17:37:46 
[LOG] 0:00.103 WARNING juju.testing failed to start mongo: mongod failed to listen on port 43220
!!build!!

File: acceptancetests/jujupy/k8s_provider/base.py [hunk 1]
```diff
@@ -169,6 +170,8 @@ def add_k8s(self, is_local=False, juju_home=None):
                 '--local',
                 '--cluster-name', self.kubeconfig_cluster_name,
             )
+            # use this local cloud to bootstrap later.
+            self.bs_manager.client.env._cloud_name = self.cloud_name
```
Comment (babbageclunk) [inline acceptancetests/jujupy/k8s_provider/base.py]: This seems a bit like reaching deep into the internals of client.env - maybe add a method to it to allow setting the cloud name? Not a biggie, but might save some confusion later on.

Comment (ycliuhw) [reply acceptancetests/jujupy/k8s_provider/base.py]: right, added set_cloud_name method in `JujuData`.

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**